### PR TITLE
Implemented local datasources with Room

### PR DESCRIPTION
**Pull request by mshaposhnyk**

### CODE REVIEW

File: app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            .subscribeOn(Schedulers.io())
+            .observeOn(Schedulers.io())
+    }
+
+    override fun getRepoPulls(repo: Repo): Single<List<Pull>> {
+        return pullDao.getPullsWithAssignees(repo.url)
+            .toObservable()
+            .flatMap {
+                Observable.fromIterable(it)
+            }
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt]: Let's change to 
`.flattenAsObservable { it }`

File: app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+
+class RoomIssueDataSource(private val issueDao: IssueDao, private val userDao: UserDao) :
+    LocalIssueDataSource {
+    override fun addIssue(issue: Issue): Completable {
+        return issueDao.addIssue(issue.fromDomain())
+            .subscribeOn(Schedulers.io())
+            .observeOn(Schedulers.io())
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt]: Why do you use `Schedulers.io()` in `observeOn()` ? Are you not going to do anything in a main thread? For example some kind of notification of success or error

File: app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    override fun getRepoIssues(repo: Repo): Single<List<Issue>> {
+        return issueDao.getIssuesWithAssignees(repo.url)
+            .toObservable()
+            .flatMap {
+                Observable.fromIterable(it)
+            }
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt]: Let's change to
`.flattenAsObservable { it }`

File: app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                if (pull.pullRoomEntity.assigneeId != null) {
+                    userDao.getUserById(pull.pullRoomEntity.assigneeId)
+                        .toObservable()
+                        .flatMap {
+                            pull.pullRoomEntity.assignee = it
+                            Observable.just(pull)
+                        }
+                } else {
+                    Observable.just(pull)
+                }
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt]: You can remove else
```
} 
 Observable.just(pull)
                
```

File: app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt (hunk 2 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt]: Changed oberveOn to AndroidSchedulers.mainThread()

File: app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt (hunk 1 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt]: Done

File: app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt (hunk 3 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/issue/RoomIssueDataSource.kt]: Done

File: app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt (hunk 4 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/db/pull/RoomPullDataSource.kt]: Done

**Pull request merged**